mitodl/open-discussions
# Return search term suggestions

#### Pre-Flight checklist


#### What are the relevant tickets?
#2506 
#2507

#### What's this PR do?
- Removes fuzziness from course search
- Returns a suggested search term if the number of results is <= X (default 1)
- Refactors the search code, so that separate functions are called for learning resource search vs discussions search

#### How should this be manually tested?
- You should have a decent number of imported courses (like edx).
- Recreate the index (there are changes to the analyzers and mappings).
- Search for 'enginr', you should get 0 results back and "Did you mean engineer" should be displayed.
- Click on the "engineer" link, it should rerun the search with that term.
- Try again with "mechical enginr", the suggestion should be "mechanical engineer"
- Run searches on posts, comments, and profiles, they should work as before (and no suggestions).

#### Screenshots (if appropriate)
in comments below
